lib/hooks/core/hook-http.js

```diff
@@ -68,11 +68,9 @@ function requestWrap(request) {
         numBytes += chunk.length;
       });
       res.on('end', function() {
-        currContext.span.setLabel(
-          TraceLabels.HTTP_RESPONSE_SIZE_LABEL_KEY, numBytes);
-        currContext.span.setLabel(
-          TraceLabels.HTTP_RESPONSE_CODE_LABEL_KEY, res.statusCode);
-        currContext.close();
+        span.addLabel(TraceLabels.HTTP_RESPONSE_SIZE_LABEL_KEY, numBytes);
+        span.addLabel(TraceLabels.HTTP_RESPONSE_CODE_LABEL_KEY, res.statusCode);
+        span.end();
```

Comment:
I don't think we have an `agent.endSpan` function. It is called in `index.js` but never defined.